src/content-tags-drawer/ContentTagsDrawer.jsx

```diff
@@ -109,6 +109,40 @@ const ContentTagsDrawer = ({ id, onClose }) => {
   }, []);
 
   const taxonomies = useMemo(() => {
+    const sortTaxonomies = (taxonomiesList) => {
+      const taxonomiesWithData = taxonomiesList.filter(
+        (t) => t.contentTags.length !== 0,
+      );
+
+      // Count implicit tags per taxonomy
+      const tagsCountBytaxonomy = {};
+      taxonomiesWithData.forEach((tax) => {
+        const countedTags = [];
+        let tagsCounter = 0;
+        tax.contentTags.forEach((tag) => {
+          tag.lineage.forEach((value) => {
+            if (!countedTags.includes(value)) {
+              countedTags.push(value);
+              tagsCounter += 1;
+            }
+          });
+        });
+        tagsCountBytaxonomy[tax.id] = tagsCounter;
+      });
+
+      // Sort taxonomies with data by implicit count
+      const sortedTaxonomiesWithData = taxonomiesWithData.sort(
+        (a, b) => tagsCountBytaxonomy[b.id] - tagsCountBytaxonomy[a.id],
+      );
+
+      // Sort empty taxonomies by name
+      const emptyTaxonomies = taxonomiesList.filter(
+        (t) => t.contentTags.length === 0,
+      ).sort((a, b) => a.name.localeCompare(b.name));
+
+      return [...sortedTaxonomiesWithData, ...emptyTaxonomies];
+    };
```

Comment:
I think this code can be simplified quite a bit. If I understand correctly, you want to sort first by the tag count and then by the name? The complication being that all implicit tags need to be counted as well. 

I see that this value is being displayed in the UI, so can't that value be cached and reused? Why is it being recalculated for the sorting here? 

I think javascript sorting is stable, so if the API response is already sorted by name you can just sort by tag count here, and it should sort the whole list by tags while retaining the order of the other elements. 

So the whole tag count code can be replaced with roughly:

`taxonomiesList.map(item => (new Set(item.contentTags.flatMap(item=>item.lineage))).size)`

This will create a flat list of all lineage items, convert that to a set, so that duplicates are removed an then get the size. You then have a list of tag counts and can sort by this list.

Reply:
Thanks for the feedback! I updated the code here https://github.com/openedx/frontend-app-course-authoring/pull/949/commits/9ded66ee5d541fd9f14a02a1ba4083ad4d03cf96

> I see that this value is being displayed in the UI, so can't that value be cached and reused? Why is it being recalculated for the sorting here?

Yes, you are right, that calculation is done individually here: https://github.com/openedx/frontend-app-course-authoring/blob/08140226c3d3601378f36215f366826557b9023e/src/content-tags-drawer/ContentTagsCollapsibleHelper.jsx#L140-L176

For reasons of complexity and budget I have left a comment about this, to do it in the future. The good thing is that with or without the counting the function will be executed so it does not affect performance.